File: Jarvis.Web/MG.Jarvis.Web/ClientApp/src/app/extranet/viewmodel/roomtypeviewmodel/hotelroomtypeviewmodel.ts
import { IObjectWithState } from '../../../common/iobjectwithstate';
import { ObjectState } from '../../../common/enums';
import { RoomBedOptionsViewModel } from './roombedoptionsviewmodel';

export class HotelRoomTypeViewModel implements IObjectWithState {
        objectState: ObjectState;
        roomId: Number;
        isActive: boolean;
        hotelId: number;
        roomTypeId: number;
        name: string;
        size: number;
        sizeMeasureId: number;
        noOfRooms: number;
        isSmoking: boolean;
        isFreeSale: boolean;
        description: string;
        roomBedOptions: RoomBedOptionsViewModel;
}

File: Jarvis.Web/MG.Jarvis.Api.Extranet/ViewModel/HotelRoomTypeViewModel.cs
using System.ComponentModel.DataAnnotations;

namespace MG.Jarvis.Api.Extranet.ViewModel
{
    public class HotelRoomTypeViewModel:BaseViewModel
    {
        public int RoomId { get; set; }
      
        public bool IsActive { get; set; }
        [Required]
        public new int? HotelId { get; set; }
        [Required]
        public int? RoomTypeId { get; set; }
        [Required]
        public string Name { get; set; }
        [Required]
        public int? Size { get; set; }
        [Required]
        public int? SizeMeasureId { get; set; }
        [Required]
        public int? NoOfRooms { get; set; }
        public int NoOfTwinRooms { get; set; }
        public int NoOfDoubleRooms { get; set; }
        [Required]
        public bool? IsSmoking { get; set; }
        public bool? IsFreeSale { get; set; }
        [Required]
        public string Description { get; set; }
        [Required]
        public RoomBedOptionViewModel RoomBedOptions { get; set; }
    }
}
